File: Server/models/Trip.js
const mongoose = require('mongoose');
const Counter = require('./Counter');

const tripSchema = new mongoose.Schema({
  trip_id: {
    type: Number,
    unique: true
  },
  user_id: {
    type: Number,
    required: true,
    ref: 'User',
    index: true
  },
  destination: {
    type: String,
    required: true
  },
  start_date: {
    type: Date,
    required: true
  },
  end_date: {
    type: Date,
    required: true
  },
  trip_type: {
    type: String,
    default: "Du lịch"
  },
  transport: {
    type: String,
    default: "Máy bay"
  },
  image_url: {
    type: String,
    default: "https://i.postimg.cc/9MXZHYtp/3.jpg"
  },
  total_items: {
    type: Number,
    default: 0
  },
  packed_items: {
    type: Number,
    default: 0
  },

  outfit_schedule: {
  type: [
    {
      day: Number,
      outfit_id: Number,
      outfit_image: String
    }
  ],
  default: []
}
}, {
  timestamps: true
});

// Index
tripSchema.index({ user_id: 1, start_date: 1 });


tripSchema.pre('save', async function () {
  const doc = this;

  if (!doc.isNew) return;

  const counter = await Counter.findByIdAndUpdate(
    { _id: 'trip_id' },
    { $inc: { seq: 1 } },
    { new: true, upsert: true }
  );

  doc.trip_id = counter.seq;
});
console.log("NEW TRIP MODEL LOADED");


module.exports = mongoose.model('Trip', tripSchema, 'trips');

File: Android/app/src/main/java/com/example/smartfashion/model/Trip.kt
package com.example.smartfashion.models

import com.google.gson.annotations.SerializedName

data class Trip(
    @SerializedName("_id")
    val id: String,

    @SerializedName("trip_id")
    val tripId: Int,

    @SerializedName("user_id")
    val userId: Int,

    val destination: String,

    @SerializedName("start_date")
    val startDate: String,

    @SerializedName("end_date")
    val endDate: String,

    @SerializedName("trip_type")
    val tripType: String,

    val transport: String,

    @SerializedName("image_url")
    val imageUrl: String?,

    @SerializedName("total_items")
    val totalItems: Int,

    @SerializedName("packed_items")
    val packedItems: Int
)
